app/assets/javascripts/controllers/toolbar_controller.js

```diff
@@ -143,9 +143,12 @@
   }
 
   ToolbarController.prototype.onUpdateItem = function(updateData) {
-    _.find(_.flatten(this.toolbarItems), {id: updateData.update})[updateData.type] = updateData.value;
-    if(!this.$scope.$$phase) {
-      this.$scope.$digest();
+    var toolbarItem = _.find(_.flatten(this.toolbarItems), {id: updateData.update});
+    if (toolbarItem && toolbarItem.hasOwnProperty(updateData.type)) {
+      updateData.type[updateData.type] = updateData.value;
```

Comment:
`updateData.type[updateData.type]` .. should be `toolbarItem[updateData.type]` instead?